source/index.js

```diff
@@ -134,6 +134,11 @@ const createStyler = (open, close, parent) => {
 
 const createBuilder = (self, _styler, _isEmpty) => {
 	const builder = (...arguments_) => {
+		if (Array.isArray(arguments_[0])) {
```

Comment:
```suggestion
		if (Array.isArray(arguments_[0]) && arguments_[0].raw) {
```

Reply:
Otherwise, this would break any sort of usecase where people are relying on array -> string coercion.

Reply:
^ Would be good to have a test for this too.

Reply:
It migh also be a good idea to cache `Array.isArray` in a module scope `const` to minimise `[[Get]]` lookups of a mutable property:
```suggestion
		if (ArrayIsArray(arguments_[0]) && ArrayIsArray(arguments_[0].raw)) {
```

Reply:
- I added a small test to see if chalk correctly casts int and array to string. This revealed a previously existing bug: `chalk(["abc"])` threw an error. This should now be fixed by adding the suggestion on line 198  as well https://github.com/chalk/chalk/blob/26c73d2b80f06f5673a7bb44433f69316f229629/source/index.js#L198
- Caching Array.isArray is indeed faster. (from 24M ops/s to 25M ops/s for the first benchmark).